src/MaaCore/Task/Miscellaneous/StartGameTaskPlugin.cpp

```diff
@@ -4,54 +4,50 @@
 
 using namespace asst;
 
+bool StartGameTaskPlugin::start_game_with_retries(size_t pipe_data_size_limit, bool newer_android)
+    const
+{
+    int extra_runs = 0;
+    for (int i = 0; i < 30; ++i) {
```

Comment:
I think we need `need_exit()` in any loop

Reply:
I can easily add it with a if and a return, but is automatic? Or we do need to add more logic to tell him when the need_exit is needed?

Reply:
> I can easily add it with a if and a return, but is automatic? Or we do need to add more logic to tell him when the need_exit is needed?
> 
> Tbh, at that point if we know a condition where we can stop the loop, we can just put it in the loop condition, instead of using need_exit()

The core will be notified when the ui hits stop, and need_exit() will return the information.

Reply:
Oh, interesting. Well considering it's a stop, we can add it exactly as the first step of the loop, to stop it as fast as possible.